taggit/managers.py

```diff
@@ -203,9 +206,31 @@ def add(self, *tags):
         for new_tag in tags_to_create:
             tag_objs.add(self.through.tag_model().objects.create(name=new_tag))
 
+        new_ids = set(t.pk for t in tag_objs)
+
+        # NOTE: can we hardcode 'tag_id' here or should the column name be got
+        # dynamically from somewhere?
+        vals = (self.through._default_manager.using(db)
+                .values_list('tag_id', flat=True)
+                .filter(**self._lookup_kwargs()))
+
+        new_ids = set(new_ids - set(vals))
+
+        signals.m2m_changed.send(
+            sender=self.through, action="pre_add",
+            instance=self.instance, reverse=False,
```

Comment:
I think it is safe to hard code `reverse=False` here as as I believe `_TaggableManager` is always used as a forward many-to-many descriptor, however, confirmation of this would be great.